python/ngen_cal/src/ngen/cal/ngen_hooks/ngen_output.py

```diff
@@ -0,0 +1,83 @@
+from ngen.cal import hookimpl
+
+import pandas as pd
+from pandas import DataFrame, Series
+from pathlib import Path
+
+class TrouteOutput():
+
+    def __init__(self, filepath: Path) -> None:
+        self._output_file = filepath
+        self._type = filepath.suffix
+    
+    def _get_dataframe(self) -> DataFrame | None:
+        """Get the t-route output raw dataframe from either csv or hdf5 files
+
+        Returns:
+            DataFrame: _description_
+        """
+        if self._type == '.csv':
+            df = pd.read_csv(self._output_file, index_col=0)
+            df.index = df.index.map(lambda x: 'wb-'+str(x))
+            tuples = [ eval(x) for x in df.columns ]
+            df.columns = pd.MultiIndex.from_tuples(tuples)
+        elif self._type == '.hdf5':
+            df = pd.read_hdf(self._output_file)
+            df.index = df.index.map(lambda x: 'wb-'+str(x))
+            df.columns = pd.MultiIndex.from_tuples(df.columns)
+        else:
+            df = None
+        return df
+    
+    # Try external provided output hooks, if those fail, try this one
+    # this will only execute if all other hooks return None (or they don't exist)
+    @hookimpl(specname="ngen_cal_model_output", trylast=True)    
+    def getOutput(self, id: str) -> Series:
```

Comment:
nitpick
```suggestion
    def get_output(self, id: str) -> Series:
```

Reply:
changed in c33327b72e1c60a42e03cc1d2f7d22f6e745dafe